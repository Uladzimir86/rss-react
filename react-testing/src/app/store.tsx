import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from './reducer';

export interface RootState {
  articles: [],
  error: {
    err: false, 
    errMessage: '',
  },
}

const store = createStore(
  reducer,
  composeWithDevTools(),
);
export default store;