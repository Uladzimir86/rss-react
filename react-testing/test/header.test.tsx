import React from 'react';
import {render as rtlRender, screen} from '@testing-library/react';
import {BrowserRouter as Router} from "react-router-dom";

import { Provider } from 'react-redux';
import store from '../src/app/store';
import Header from '../src/components/header';


const render = (ui: any, {route = '/'} = {}) => {
  window.history.pushState({}, 'Test page', route)
  return rtlRender(ui, {wrapper: Router})
}
const spy = jest.spyOn(React, 'useState');

test('should match snapshot', () => {
  const {container} = render(<Provider store={store}><Header/></Provider>);
  expect(container).toMatchSnapshot();
})
test('should set chosen link (/about)', () => {
  const route = '/about';
  render(<Header />, {route});
  expect(spy).toBeCalled();
})
test('should set chosen link (/details/)', () => {
  const route = '/details/';
  render(<Header />, {route});
  expect(spy).toBeCalled();
})
test('should set chosen link (/404)', () => {
  const route = '/404';
  render(<Header />, {route});
  expect(spy).toBeCalled();
})
 


