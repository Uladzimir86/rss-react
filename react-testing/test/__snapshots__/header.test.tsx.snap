// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="header__nav"
    >
      <ul
        class="nav__list"
      >
        <li
          class="nav__item nav__item_chosen"
        >
          <a
            class="nav__link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          class="nav__item"
        >
          <a
            class="nav__link"
            href="/about"
          >
            <span>
              About
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
